Sources/Plasma/PubUtilLib/plProgressMgr/plProgressMgr.cpp

```diff
@@ -230,15 +235,12 @@ void    plProgressMgr::CancelAllOps( void )
 
 const ST::string plProgressMgr::GetLoadingFrameID(int index)
 {
-    if (index < LOADING_RES_COUNT)
-        return fImageRotation[index];
-    else
-        return fImageRotation[0];
+    return fImageRotation[index % (fImageRotation.size() - 1)];
```

Comment:
`%` doesn't protect against negative indexes...  I'd suggest either making this unsigned or keeping the old behavior (with the non-hard-coded size)

Reply:
I must have been thinking of this backwards.  Thanks.

Reply:
The old method doesn't protect against negatives, either.  Yikes.

Reply:
Is std::clamp available for our usage?  Would that be appropriate?

Reply:
`std::clamp` was added in C++17--we are still using C++14.

Reply:
Ah, that's out then. :)

Reply:
> The old method doesn't protect against negatives, either. Yikes.

Ah, good point...  Maybe not a big deal then.